brainiak/reprsimil/brsa.py

```diff
@@ -0,0 +1,1118 @@
+        (lower-triangular matrix), shape=[condition,condition].
+    C_: the correlation matrix derived from the shared covariance matrix,
+        shape=[condition,condition]
+    nSNR_ : array, shape=[voxels,]
+        The pseuso-SNR of all voxels.
+        They are normalized such that the geometric mean is 1
+    sigma_ : array, shape=[voxels,]
+        The estimated standard deviation of the noise in each voxel
+        Assuming AR(1) model, this means the standard deviation
+        of the refreshing noise.
+    rho_ : array, shape=[voxels,]
+        The estimated autoregressive coefficient of each voxel
+    lGPspace_ : number, only if GP_space or GP_inten is True
+        the length scale of Gaussian Process prior of log(SNR)
+    bGP_ : number, only if GP_space or GP_inten is True.
+        the standard deviation of the GP prior
+    lGPinten_: number, only if GP_inten is True
+        the length scale in fMRI intensity of the GP prior of log(SNR)
+
+    Notes
+    -----
+    The current version assumes noise is independent across voxels.
+    Real data typically has spatial correlation in noise.
+    This assumption might still introduce some bias in the result.
+    Spatial correlation will be included in a future version.
+    """
+
+    def __init__(
+            self, n_iter=50, rank=None, GP_space=False, GP_inten=False,
+            tolx=2e-3, verbose=False, pad_DC=False, epsilon=0.001,
+            space_smooth_range=None, inten_smooth_range=None,
+            inten_weight=0.1, init_iter=20):
+        self.n_iter = n_iter
+        self.rank = rank
+        self.GP_space = GP_space
+        self.GP_inten = GP_inten
+        self.tolx = tolx
+        self.verbose = verbose
+        self.pad_DC = pad_DC
+        self.epsilon = epsilon
+        # This is a tiny ridge added to the Gaussian Process
+        # covariance matrix template to gaurantee that it is invertible.
+        # Mathematically it means we assume that this proportion of the
+        # variance is always independent between voxels for the log(SNR2).
+        self.space_smooth_range = space_smooth_range
+        self.inten_smooth_range = inten_smooth_range
+        self.inten_weight = inten_weight
+        # The kernel of the Gaussian Process is a weighting of a kernel
+        # defined on spatial coordinate and a kernel defined on
+        # image intensity. We think it should be primarily based on
+        # spatial coordinate. So the default weight for the kernel
+        # defined on image intensity is 0.1.
+        self.init_iter = init_iter
+        # When imposing smoothness prior, fit the model without this
+        # prior for this number of iterations.
+        return
+
+    def fit(self, X, design, scan_onsets=None, coords=None,
+            inten=None, inten_weight=None):
+        """Compute the Bayesian RSA
+
+        Parameters
+        ----------
+        X: 2-D numpy array, shape=[time_points, voxels]
+            In the equation, this is actually Y.
```

Comment:
This is confusing because of the shift from generative to descriptive perspective -- but I am not sure what the best solution is. At the very least, the derivation that ships with brainiak should match notation with the code and also show the relationship to the NIPS paper notation. But maybe there is a better option, maybe someone else can think of it?